drivers:platform:maxim:max32670 Initial implementation of GPIO platform drivers

mphalke opened the pull request:
The GPIO drivers implementation is referred from the MAX32670 folder as both parts of identical GPIO implementation.
Few minor file edits done w.r.t. code styles and comments but the actual functionality is unaltered.

Signed-off-by: mahphalke <Mahesh.Phalke@analog.com>

CiprianRegus commented on drivers/platform/maxim/max32670/maxim_gpio.c:
```diff
@@ -0,0 +1,343 @@
+
+	pextra = param->extra;
+
+	switch(param->pull) {
```
Perhaps it would be better if we also add the `NO_OS_PULL_UP_WEAK` and `NO_OS_PULL_DOWN_WEAK` as fallthrough cases to `NO_OS_PULL_UP` and `NO_OS_PULL_DOWN` if the max32670 doesn't support setting more than one resistance value for the pull ups.

mphalke replied:
Make sense.

mphalke commented on drivers/platform/maxim/max32670/maxim_gpio.c:
```diff
@@ -0,0 +1,343 @@
+
+	pextra = param->extra;
+
+	switch(param->pull) {
```
Will fix at all spaces

mphalke commented:
V1.0:
Review comment fixes:

- Added NO_OS_PULL_UP_WEAK and NO_OS_PULL_DOWN_WEAK as fall-through cases to NO_OS_PULL_UP and NO_OS_PULL_DOWN
- Added extra space between switch statement and opening bracket.
- Other minor code edits as per review comments.
